app/editor/src/hooks/api-editor/editor/useApiHub.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  HubConnection,
+  HubConnectionBuilder,
+  HubConnectionState,
+  LogLevel,
+} from '@microsoft/signalr';
+import React from 'react';
+import { useAppStore } from 'store/slices';
+import { useKeycloakWrapper } from 'tno-core';
+
+import { HubMethodName, Settings } from '../constants';
+
+const url = Settings.ApiPath + '/hub';
+let hub: HubConnection;
```

Comment:
Use a module singleton variable so that it's one connection for the whole application.